src/rgw/rgw_rados.cc

```diff
@@ -9400,6 +9403,30 @@ int RGWRados::cls_obj_usage_log_read(const string& oid, const string& user, cons
   return r;
 }
 
+int RGWRados::cls_rgw_usage_log_trim_rgw_rados_operate(rgw_rados_ref ref, const string& user, const string& bucket, uint64_t start_epoch, uint64_t end_epoch)
```

Comment:
@cbodley this should get to the heart of what we discussed about `cls_rgw_log_trim`.

Reply:
lets drop the `_rgw_rados_operate` part of the function name

also, you declared an `ObjectWriteOperation` overload of `cls_rgw_usage_log_trim()` in cls_rgw_client.h but i don't see its definition in cls_rgw_client.cc. you should be able to call that inside this do-while loop instead of encoding the bufferlist manually - that way if we change something in cls/rgw, we won't break this part
```c++
  do {
    librados::ObjectWriteOperation op;
    cls_rgw_usage_log_trim(op, user, bucket, start_epoch, end_epoch);
    int r = rgw_rados_operate(ref.ioctx, ref.obj.oid, &op, null_yield);
```